[Shanyey] IP

Shanyey opened the pull request:
# Nova Chatbot 😎 
This chat bot is 
- **Motivational**
- <sub>Engaging</sub>
- ~~Easy to use~~

Wonder how Nova can help you keep track of your tasks? Fret not, here are what it can do!
1. Store tasks like Todo, deadline, and events
2. mark and unmark tasks to keep track of your progress
3. Filter to find specific tasks you are looking for

Features
[x] Managing tasks
[ ] Reminders
[ ] GUI 

Here is the main code if you are interested!
```ruby

public static void main(String[] args) {
        new Nova().run();
    }

```

> [!WARNING]
> This project is a work-in-progress
If you wish to know more, please visit this [website](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Shanyey committed:
level-0: rename, Greet, Exit

Shanyey committed:
Edit variable name

Shanyey committed:
Echo

Shanyey committed:
Level-2: add, list

Shanyey committed:
Level 3: mark as done

Shanyey committed:
Halfway done with Deadline class

Shanyey committed:
Level-4

Shanyey committed:
add testing

Shanyey committed:
shift classes to classes folder
add NovaException

Shanyey committed:
add classes folder
add NovaException
update ReadMe

Shanyey committed:
add personality

Shanyey committed:
optimise code

Shanyey committed:
optimise code
add error testing for mark, unmark

Shanyey committed:
add error test

Shanyey committed:
add delete functionality with error checking

Shanyey committed:
add save and load function

Shanyey committed:
Merge branch 'branch-Level-7'

Shanyey committed:
branch Level 8

Shanyey committed:
Merge branch 'branch-Level-8'

Shanyey committed:
add storage class

Shanyey committed:
add more OOP

Shanyey committed:
Merge tag 'A-MoreOOP'

Shanyey committed:
set up packages

zhenglong1603 commented on src/main/java/nova/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+package nova.storage;
+
+import java.io.*;
```
consider adjusting this since imported classes should be stated explicitly?
![image](https://github.com/user-attachments/assets/c524489c-2963-4e68-bac3-f9ad5c94429d)

Shanyey replied:
Fixed!

zhenglong1603 commented on src/main/java/nova/taskList/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+        }
+        Task task = tasks.remove(index - 1);
+        System.out.println("I knew you wouldn't do this task" + "\n" + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
```
maybe consider abstracting such responses and letting the UI class do all the generating of the responses instead?

Shanyey replied:
Thank you for the suggestion!

Shanyey committed:
Merge branch 'add-gradle-support'

zhenglong1603 commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,61 @@
+
+public class Nova {
+    private static final Scanner scanner = new Scanner(System.in);
+    //private static ArrayList<Task> tasks;
```
is this needed since it is commented out ?

Shanyey replied:
It is indeed not needed

zhenglong1603 commented on taskData.txt:
maybe consider adding this file to your gitignore as well to keep the repo cleaner !

zhenglong1603 commented on gradle/wrapper/gradle-wrapper.properties:
consider adding the gradle file to gitignore for devops purposes as well :)

zhenglong1603 commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,61 @@
+                try {
+                    String[] splitAction = parser.parseBySpace(action); //[ type, desc + others]
+                    String[] slashedAction = parser.splitBySlash(splitAction[1]); //[desc, by or start, to]
+                    switch (splitAction[0].toLowerCase()) {
```
take note of the indentation for switch case!
you can take a look at this!
https://se-education.org/guides/tutorials/intellijCodeStyle.html

zhenglong1603 reviewed (commented):
Noticed that you are still working on the tasks from last week prob because of the CNY break:) Press on and hope to see you catch up and start implementing the tasks this week as well:)

Shanyey committed:
add gradle support

Shanyey committed:
add JUNIT test

Shanyey committed:
add javadoc comments

Shanyey committed:
A-Coding Standard

Shanyey committed:
Merge branch 'A-JavaDoc'

Shanyey committed:
Merge branch 'A-JavaDoc' into A-CodingStandard

# Conflicts:
#	src/main/java/nova/Nova.java
#	src/main/java/nova/storage/Storage.java

Shanyey committed:
fix merge conflicts

Shanyey committed:
fix bugs

Shanyey committed:
add Find functionality

Shanyey committed:
Merge tag 'Level-9'

Shanyey committed:
add checkstyle

Shanyey committed:
Merge branch 'A-CheckStyle'

awlarpi commented on src/main/java/nova/tasks/Event.java:
```diff
@@ -0,0 +1,74 @@
+     * @param by The end date/time of the event in "yyyy-MM-dd HH:mm" format.
+     * @param isDone A boolean indicating whether the event is completed.
+     */
+    public Event(String description, String startingDate, String by, Boolean isDone) {
```
Nice use of overloading!

awlarpi commented on src/test/java/nova/parser/ParserTest.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    public void parseBySpace_success() {
```
Perhaps you can consider adding test cases for parsing other types of input, not just todos.

awlarpi commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,83 @@
+public class Nova {
+    private static final Scanner scanner = new Scanner(System.in);
+
+    private final Storage storage;
+    private final TaskList tasks;
+    private final Parser parser;
+    private final TextUi textUi;
```
Good use of the final keyword!

awlarpi reviewed (commented):
Looks good to me, the classes are well designed. Maybe you can consider using the Command pattern to handle the commands, at least that's what I did for my code.

awlarpi commented on src/main/java/nova/storage/Storage.java:
```diff
@@ -0,0 +1,80 @@
+ * Tasks are stored in "taskData.txt" and retrieved when the program starts.
+ */
+public class Storage {
+    private static final String FILE_NAME = "taskData.txt";
```
Maybe can consider handling the case that the user somehow accidentally deletes or creates another file called taskData.txt without knowing that it is actually supposed to be reserved for the Nova program.

Shanyey committed:
add GUI functionality

Shanyey committed:
Merge branch 'Level-10'

Junixm commented on src/main/java/nova/command/Command.java:
```diff
@@ -0,0 +1,146 @@
+import nova.tasklist.TaskList;
+
+public class Command {
+    TaskList tasklist;
+    Parser parser;
```
LGTM

Junixm commented on src/main/java/nova/nova/Nova.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Nova {
+
+    private final Storage storage;
+    private final TaskList tasks;
+    private final Parser parser;
+    private final Command command;
```
LGTM

Junixm commented on src/main/java/nova/tasks/Event.java:
```diff
@@ -0,0 +1,74 @@
+ * @author shanyey
+ */
+public class Event extends Task {
+    protected String by;
+    protected String startingDate;
```
change by to endingDate to improve readability

Junixm commented on src/main/java/nova/tasks/Event.java:
```diff
@@ -0,0 +1,74 @@
+public class Event extends Task {
+    protected String by;
+    protected String startingDate;
+    protected LocalDateTime start;
+    protected LocalDateTime end;
```
change it to startTime and endTime would be better

Shanyey committed:
add error handling of empty command input

In nova, handle user pressing enter without typing any commands in to
the text box.

Shanyey committed:
add assertion

Added assertion checks to command and parser to check for input's
validity

Shanyey committed:
fix code quality

I removed UI as it is no longer needed. Extracted out methods in
tasklist, command, to follow SLAP. Added new method in task to check if
its description contains a given string

Shanyey committed:
Merge pull request #2 from Shanyey/branch-A-Assertions

add assertion

Shanyey committed:
Merge branch 'master' into branch-A-CodeQuality

Shanyey committed:
Merge pull request #3 from Shanyey/branch-A-CodeQuality

fix code quality

Shanyey committed:
Merge branch 'master' of https://github.com/Shanyey/ip

* 'master' of https://github.com/Shanyey/ip:
  fix code quality

Shanyey committed:
add undo and redo function

Created a new class called HistoryManager to store data about the
array list.

Shanyey committed:
Merge branch 'branch-BCD-Extension'

Shanyey committed:
adjust gui

added more gui features and help function

Shanyey committed:
add personality and code quality

added special error messages, some refactoring to extract and improve
abstraction

Shanyey committed:
add more error handling

Shanyey committed:
Add JUnit test for core classes

The project lacks testing for critical modules like Storage or TaskList.

Without test coverage, changes may introduce regressions and
manual testing becomes unreliable and error-prone.

Add comprehensive JUnit 5 test classes to validate functionality and
 error handling for these components.

This is done to ensure that key behaviors are verified automatically,
 increasing stability and maintainability of the codebase.

Shanyey committed:
Add user guide

There is no explanation on how to use the app

Users who try the app must be able to learn how to operate the chatbot.

Add user guide to docs folder

Shanyey committed:
add javadoc comments and fix readme

There were less than half of the methods with proper javadoc comments.
Readme was not as clear and had redundant items.

Shanyey committed:
remove original readme

Shanyey committed:
fix readme

Shanyey committed:
edit invalid command message

Shanyey committed:
update user guide

Shanyey committed:
Fix formatting
